### ISSUE TITLE
`func` should be allowed in type descriptions

### ISSUE THREAD

**Issue by khogeland** [nim-lang/Nim#11518]
If you're using `func`, you're probably writing functional code that is frequently going to have functors as arguments or fields, where you now have to use `proc(): foo {.noSideEffect.}`. The utility seems lost if you still need to use the pragma in type descriptions.

Is there a better fix for #10838 that would allow `func` in type descriptions? If not, is `func` really still useful?

**Comment by Araq** [comment]
Are you really asking if proc/func declarations are more common than callback type definitions? Yes, they are!

**Comment by khogeland** [comment]
I didn't say that. Supporting `func` in type descriptions makes it easier to write functional code where higher-order functions are common.

**Comment by Araq** [comment]
Alright.

**Comment by Grinshpon** [comment]
Has there been any progress on this?

**Comment by Araq** [comment]
No. But the workaround is ``type P = proc (args...) {.noSideEffect.}``

**Comment by Grinshpon** [comment]
I made a small feature request (#13947) and a pr (#13948) that adds two macro operators to the sugar module in the standard library, which allows you to easily create side effect free function types and lambdas. It's a convenient workaround I think.
